### PR TITLE
Quick offer edit actions collapsable in company offers management widget

### LINKED ISSUE #305: Extend offer publish date
Companies should be able to quickly extend an offer's publish period by a fixed amount, preferably using a "quick-action button".

### PR DESCRIPTION
**Pull request by tomaspalma**
Closes #305 

# What was done

- Show collapsable with fields that are supposed to be quick edited (for now it's just the publish end date)

- It already updates the publish end date upon selection of date and clicking on the submit button

- It already updates the row value of the publish end date

The only thing different than in the video is the fact that now instead of the text saying there is an error or "sucessfully updated", there is a snackbar.

# Desktop View

https://user-images.githubusercontent.com/59887569/235507613-c4ccad5d-8fc1-436f-a3cc-485b36a00fd7.mp4

# Mobile View

https://github.com/NIAEFEUP/nijobs-fe/assets/59887569/4b3a7a65-0114-43b1-97bc-87f9ea6008b3

# Explanations

There was a lot of difficulty on establishing a direct relationship between the offer row and the collapsed quick edit because of the way the code is structured.

### How is the offer publish end date being updated?

The ```CompanyOffersManagementWidget``` component has a local state called ```offers```.

That state and its setter are passed to the newly created ```CollapsedQuickOfferEdit``` component.

When the publish end date is altered in the ```CollapsedQuickOfferEdit```, this component changes the offer object pointing to the respective id of the offer, changes the value of the publishEndDate and it re-renders the component.

### CODE REVIEW

File: src/components/Company/Offers/Manage/CollapsedQuickOfferEdit.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                        variant="contained"
+                        color="primary"
+                        type="submit"
+                        data-testid="submit-offer"
+                    >
+                        Submit
+                    </Button>
+                    {success &&
+                        <Typography>
+                            Sucessfully updated
```
Comment (dsantosferreira) [inline src/components/Company/Offers/Manage/CollapsedQuickOfferEdit.js]: Small typo

```suggestion
                            Successfully updated
```

File: src/components/Company/Offers/Manage/CollapsedQuickOfferEdit.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    const isObjectEmpty = useCallback(
+        (object) =>
+            Object.keys(object).length === 0 &&
+      Object.getPrototypeOf(object) === Object.prototype
+        , []);
+
+    const displaySuccessMessage = useCallback(() => {
+        dispatchAddSnackbarAction({
+            message: "Succesfully updated",
```
Comment (dsantosferreira) [inline src/components/Company/Offers/Manage/CollapsedQuickOfferEdit.js]: Small typo

```suggestion
            message: "Successfully updated",
```

**Review by dsantosferreira — APPROVED** [review]
Looks good to me! Only noticed a small typo.
Don't forget to rebase.

File: src/components/Company/Offers/Manage/CollapsedQuickOfferEdit.js (hunk 2 above)
Comment (tomaspalma) [reply src/components/Company/Offers/Manage/CollapsedQuickOfferEdit.js]: Thank you, I already rebased and fixed that minor spelling mistake.

**Review by FranciscoCardoso913 — APPROVED** [review]

**Pull request merged**